pkg/configtx/consortiums_test.go

```diff
@@ -945,6 +945,65 @@ func TestGetConsortiums(t *testing.T) {
 	gt.Expect(len(baseConsortiums)).To(Equal(len(consortiums)))
 }
 
+func TestGetConsortiumOrg(t *testing.T) {
+	t.Parallel()
+	gt := NewGomegaWithT(t)
+
+	consortiumGroup, err := baseConsortiumChannelGroup(t)
+	gt.Expect(err).NotTo(HaveOccurred())
+
+	config := &cb.Config{
+		ChannelGroup: consortiumGroup,
+	}
+
+	org1ConfigGroup, err := getConsortiumOrg(config, "Consortium1", "Org1")
+	gt.Expect(err).NotTo(HaveOccurred())
+	gt.Expect(org1ConfigGroup).To(Equal(config.ChannelGroup.Groups[ConsortiumsGroupKey].Groups["Consortium1"].Groups["Org1"]))
```

Comment:
```suggestion
	require.NoError(t, err)
	expectedConfigGroup := config.ChannelGroup.Groups[ConsortiumsGroupKey].Groups["Consortium1"].Groups["Org1"]
	require.Equal(t, expectedConfigGroup, org1ConfigGroup)
```
The above-suggested changes definitely do not hurt my eyes. 

What is the reasoning behind using `Gomega` over the `testify` package for unit tests? 

Quote the Gomega website:

> Gomega is a matcher/assertion library. It is best paired with the Ginkgo BDD test framework, but can be adapted for use in other contexts too.

Quote from ginkgo website

> Ginkgo is a Go testing framework built to help you efficiently write expressive and comprehensive tests using Behavior-Driven Development (“BDD”) style. It is best paired with the Gomega matcher library but is designed to be matcher-agnostic.

I understand the usage of Gomega with the ginkgo as we can easily search for a particular string in the stdout or stderr. This is good for integration tests of a distributed system. Why are we applying Gomega even for the UT? what is the advantage when it is hard to read the function chaining? 

Further, the testify seems to be more popular than Gomega https://go.libhunt.com/compare-testify-vs-gomega
Using testify would certainly be good for golang developers who try to contribute to Fabric. 

Comparison analysis of the golang testing framework: https://www.slideshare.net/DushyantBhalgami/comparative-analysis-of-golang-testing-frameworks 

Let's see what packages are used by existing golang projects:

1. https://github.com/etcd-io/etcd -- reflect and basic if-then-else with `t.Errorf `or `t.Fatal` from `testing` package.
2. https://github.com/docker/docker-ce -- `gotest.tools/v3/assert `and `gotest.tools/v3/assert/cmp`
3. https://github.com/hashicorp/consul -- reflect basic if-then-else with `t.Errorf` or `t.Fatal` from `testing` package.
4. https://github.com/kubernetes/kubernetes -- a combination of `github.com/stretchr/testify/assert` and basic if-then-else with `t.Errorf` or `t.Fatal `from `testing` package
5. https://github.com/golang/go --basic if-then-else with `t.Errorf` and `t.Fatal` from `testing` package
6. https://github.com/pingcap -- a combination of own assertion package `github.com/pingcap/check` and basic if-then-else with t.Errorf or t.Fatal from `testing` package
7. https://github.com/cockroachdb/cockroach -- `reflect` and basic if-then-else with `t.Errorf `or `t.Fatal`. from `testing` package

Looks like only Kubernetes uses testify and no one uses `Gomega`.  Docker and pingcap use some assertion library. 

We need to either continue with `testify` or follow basic  basic if-then-else with `t.Errorf `or `t.Fatal` from `testing` package similar to what many popular projects have done. But, PLEASE do not use Gomega for the unit test.

Reply:
While you make many valid points, I'm not sure this is the place to raise your concerns over the use of Gomega in unit tests. Perhaps you can raise these points in the fabric-maintainers channel on rocketchat if you feel strongly that we shouldn't use Gomega in any unit tests? For the time being, we've decided to use Gomega for our matchers in this package and will continue to do so to keep things consistent throughout the package. That is the same philosophy I've seen adopted in terms of mocks/fakes in fabric: stick with one type (counterfeiter v. mockery, for example) for the entire package.

Reply:
I don't want to drag it either to rocket chat or #fabric-dev. This is my code review asking reasoning for using the Gomega when there are simple ways to do the same assertion. In the ledger packages, we use simple assertions. You have provided the reasoning. Hence, please feel free to ignore my comments.  

Maybe I should ask Fabric Architect (@sykesm) that 

1. why are we not keeping UT consistent across Fabric?
2. why Gomega that is best suited with ginkgo is used for the UT when it is specifically designed for integration test with CLI?
3. why there is no guideline on this?  
4. why most of the famous existing golang projects use a simple UT framework while we have inconsistent and complex framework?
5. Do we really need `gt.Expect(org1ConfigGroup).To(BeNil())` when we can have a easily readable code such as `require.Nil(t, org1ConfigGroup)` or `if org1ConfigGroup != nil {t.Errorf(...)}`?

I am not asking for answers here. I am just asking you to think about it. If you think my points are valid, please instruct the team to avoid using Gomega in UT. Otherwise, let's use whatever we want.

Reply:
Not to speak on behalf of @sykesm or anything but we've discussed these points before in our standups/IPMs so I'll try to provide some insight on our reasoning. Simply put, I personally prefer Gomega style matchers over testify's (and I believe a majority of our squad does as well), but that's entirely my opinion. It reads better left to right like a spoken sentence in English. I wouldn't say that Gomega should explicitly be used with integration tests. Although we don't tend to use them as much in the UTs, I much rather prefer Gomega's handling for asynchronous assertions from both an integration test standpoint and when applicable in a UT point of view. In terms of consistency, there are already a couple of other places in the Fabric UT that do a mixed approach using Gomega matchers without Ginkgo. Ultimately though, because this package won't live in Fabric, it's not as clear cut as to any standardization patterns with existing Fabric code. I understand your concerns and do think maybe we should take a step back at some point and discuss at a higher level some standardization patterns/general guidelines (both for Fabric and any additional libraries we create repos for), but for now we will continue to move with Gomega matchers as the config work is nearly wrapping up in its current iteration and putting effort on converting the assertions would hold up the next set of priorities we have coming up.

Reply:
Thanks @caod123 for the response. 

1. I work remotely and sorry for not knowing about earlier discussions on UT framework.
2. In the ledger squad, we have decided to stick to assert/require as Gomega is not used by any of the famous projects that I mentioned above. We aren't using it because of that but we decided to follow a simple if-then-else or assert/require which can be read by anyone without having prior knowledge on the framework used. We prefer not to put additional learning burden for the new developers. Further Gomega follows JAVA method chaining style such as `user.setName("skptricks").setAge(22).getUserDetails();` and it does not look like a golang way to doing things to me. 
3. Testify does have support for testing with asynchronous operations. I recently used `assert.Eventually` and `assert.Never` in one of the PR. In fact, the implementation logic seems to be similar:
**Gomega**: https://github.com/onsi/gomega/blob/master/internal/asyncassertion/async_assertion.go#L105 
**testify**: https://github.com/stretchr/testify/blob/master/assert/assertions.go#L1628

Reply:
> Ultimately though, because this package won't live in Fabric. but for now we will continue to move with Gomega matchers as the config work is nearly wrapping up in its current iteration and putting effort on converting the assertions would hold up the next set of priorities we have coming up.

I agree. My goal is not to create a burden for the ongoing work. For ledger refactoring work, I have been looking at many famous open golang projects, to see what they have used, and then analyze why they have followed a certain approach over another. One of them was related to the testing framework as well. There is no single golden golang project that we can refer too. While golang repo itself is a good reference, it is more of a library repo. Hence, I am looking at multiple repos and try to identify reasoning for choosing one over the other. 

> and putting effort on converting the assertions would hold up the next set of priorities we have coming up.

Thank you much @caod123  for at least giving a consideration (even if it does not happen after some discussion).